pkg/authentication/authentication.go

```diff
@@ -0,0 +1,254 @@
+package authentication
+
+import (
+	"bytes"
+	"encoding/base64"
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/errors"
+)
+
+type encodedAuth struct {
+	Auth string `json:"auth"`
+}
+
+type allAuths struct {
+	Auths map[string]encodedAuth `json:"auths"`
+}
+
+// SetAuthentication stores the base64 encoding of username and password
+// in the auth file if the credentials are authenticated
+func SetAuthentication(username, password, server string) error {
+	path, err := getPathToAuth()
+	if err != nil {
+		return errors.Wrapf(err, "error getting path to auth file")
+	}
+
+	server = getFullURL(server)
+
+	isValid, err := checkAuthentication(username, password, server)
+	if err != nil {
+		return errors.Wrapf(err, "an error occured while authenticating credentials")
+	}
+	if isValid {
+		encodedCreds := encodeUserAndPass(username, password)
+		return addAuthToJSON(server, encodedCreds, path)
+	}
+	return nil
+}
+
+// GetAuthentication returns the value of the registry url stored in
+// the auth.json file
+func GetAuthentication(server string) (string, error) {
+	server = getFullURL(server)
+	path, err := getPathToAuth()
+	if err != nil {
+		return "", errors.Wrapf(err, "error getting path to auth file")
+	}
+
+	auths, err := readJSONFile(path)
+	if err != nil {
+		return "", errors.Wrapf(err, "error reading JSON file")
+	}
+
+	if val, ok := auths.Auths[server]; ok {
+		return val.Auth, nil
+	}
+	return "", nil
+}
+
+// getPath gets the path of the auth.json file since the user can set their own path
+// by setting CRIO_AUTH_PATH
+// If CRIO_AUTH_PATH is not set, the default path is ~/.config/containers/auth.json
+func getPathToAuth() (string, error) {
+	if os.Getenv("CRIO_AUTH_PATH") != "" {
+		return os.Getenv("CRIO_AUTH_PATH") + "/auth.json", nil
+	}
+	dir := os.Getenv("HOME") + "/.config/containers"
+	if _, err := os.Stat(dir); os.IsNotExist(err) {
+		if err = os.Mkdir(dir, os.ModePerm); err != nil {
```

Comment:
This is probably not the right set of permissions to apply to the directory.

Reply:
0600, is probably what you want.

Reply:
Fixed